SyncCalendarsIntoOne.gs

```diff
@@ -72,69 +48,97 @@ function deleteEvents(startTime, endTime) {
+
+  let requestBody_delete = [];
+
+  // get target calendar events
+  const target_events = Calendar.Events.list(CALENDAR_TO_MERGE_INTO, {
+    timeMin: startTime.toISOString(),
+    timeMax: endTime.toISOString(),
+    singleEvents: true,
+    orderBy: 'startTime',
+    }).items.filter((event) => event.summary.includes(SEARCH_CHARACTER));
 
   for (let calendarName in CALENDARS_TO_MERGE) {
+    
+    // get source calendar events
     const calendarId = CALENDARS_TO_MERGE[calendarName];
-    const calendarToCopy = CalendarApp.getCalendarById(calendarId);
-
-    if (!calendarToCopy) {
+    if (!CalendarApp.getCalendarById(calendarId)) {
       console.log("Calendar not found: '%s'.", calendarId);
       continue;
     }
-
-    // Find events
-    const events = Calendar.Events.list(calendarId, {
+    const source_events = Calendar.Events.list(calendarId, {
       timeMin: startTime.toISOString(),
       timeMax: endTime.toISOString(),
       singleEvents: true,
       orderBy: 'startTime',
     });
-
-    // If nothing find, move to next calendar
-    if (!(events.items && events.items.length > 0)) {
+    // If nothing is found, move to next calendar
+    if (!(source_events.items && source_events.items.length > 0)) {
       continue;
     }
 
-    events.items.forEach((event) => {
-      // Don't copy "free" events.
-      if (event.transparency && event.transparency === 'transparent') {
-        return;
+    // delete target event if its not in the source calendar anymore
+    target_events.forEach((target_event) => {
+      const source_event = source_events.items.filter((candidate_event) => (candidate_event.iCalUID == target_event.iCalUID))[0];  
+      if (!source_event){
+        requestBody_delete.push({
+          method: 'DELETE',
+          endpoint: `${ENDPOINT_BASE}/${CALENDAR_TO_MERGE_INTO}/events/${target_event.id}`,
+        });
       }
-
-      requestBody.push({
-        method: 'POST',
-        endpoint: `${ENDPOINT_BASE}/${CALENDAR_TO_MERGE_INTO}/events`,
-        requestBody: {
-          summary: `${SEARCH_CHARACTER}${calendarName} ${event.summary}`,
-          location: event.location,
-          description: event.description,
-          start: event.start,
-          end: event.end,
-        },
-      });
     });
-  }
-
-  if (requestBody && requestBody.length) {
-    const result = new BatchRequest({
-      batchPath: 'batch/calendar/v3',
-      requests: requestBody,
+    submitRequest(requestBody_delete, "delete")
+
+
+    // check if target event exists and update it if it does. if not, then create it.
+    requestBody_update = []
+    requestBody_create = []
+    let new_description = ""
+
+    source_events.items.forEach((source_event) => {
+      // Only consider events that are not "free".
+      if (source_event.transparency !== 'transparent') {
+
+        const target_event = target_events.filter((candidate_event) => (candidate_event.iCalUID == source_event.iCalUID))[0];
```

Comment:
Should it be `candidate_event.iCalUID === source_event.iCalUID`? What if candidate_event is null?